### PR TITLE
Add job enqueueing for domain scans

### PR DESCRIPTION
**Pull request by mimi89999**

### CODE REVIEW

File: app/templates/check_running.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% extends "custom_layout.html" %}
+
+{% block body %}
+    <div class="container">
+        <div class="card">
+            <div class="card-body">
+                <h5 class="card-title">{% trans %}Domain analysis is running{% endtrans %}</h5>
+                <div class="card-text">
+                    <img class="spinner waiting" src="/static/images/spinner.svg" />
+                    <h5 class="waiting">{% trans %}Waiting for the domain analysis to finfish{% endtrans %}</h5>
```
Comment (msm-cert) [inline app/templates/check_running.html:10]: maybe add an autmatic page refresh every 5s here (with javascript or http header or http-equiv)?

**Review by msm-cert — COMMENTED** [review]
Some CI checks are failing.

Also consider auto reload of the page?

**Comment by kazet** [comment]
We already have refresh logic in the email check feature

czw., 18 sty 2024, 18:44 użytkownik msm-cert ***@***.***>
napisał:

File: app/templates/check_running.html (hunk 1 above)
Comment (mimi89999) [reply app/templates/check_running.html:14]: Done

File: app/src/app.py [hunk 2]
```diff
@@ -129,8 +131,10 @@ async def check_email_results(request: Request, recipient_username: str) -> Resp
         error = translate("Invalid or no e-mail domain in the message From header", Language(Config.UI.LANGUAGE))
     else:
         try:
+            client_ip = request.client.host if request.client else None
+            client_user_agent = request.headers.get("user-agent", None)
+
             result = scan_and_log(
```
Comment (kazet) [inline app/src/app.py:137]: what is the reason scanning of the e-mail domain and DKIM signature is performed synchronously and scanning the domain is async?

File: app/templates/check_running.html (hunk 1 above)
Comment (kazet) [reply app/templates/check_running.html:14]: finfish -> finish

File: app/templates/check_running.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+                </div>
+            </div>
+        </div>
+   </div>
+{% endblock %}
```
Comment (kazet) [inline app/templates/check_running.html:19]: IMO it's not the most beautiful UI possibl:

image

File: app/translations/en_US/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -262,3 +270,6 @@ msgid ""
 "<b>SPF</b> and <b>DMARC</b> mechanisms will be checked. To check DKIM, "
 "you need to send a test e-mail."
 msgstr ""
+
```
Comment (kazet) [inline app/translations/en_US/LC_MESSAGES/messages.po:273]: don't leave old i18n headers

File: app/src/worker.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+def check_domain_scan_job(
+    client_ip: Optional[str],
+    client_user_agent: Optional[str],
+    envelope_domain: str,
+    token: str,
+    message_key: Optional[bytes] = None,
+    recipient_username: Optional[str] = None,
+) -> None:
+    if message_key:
```
Comment (kazet) [inline app/src/worker.py:33]: imo this function can be split into two, the "has message" and "has no message" parts

File: app/src/worker.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+from .scan import DomainValidationException, ScanningException
+from .translate import translate
+
+LOGGER = build_logger(__name__)
+REDIS = Redis.from_url(Config.Data.REDIS_URL)
+
+setup_resolver()
+
+
+def check_domain_scan_job(
```
Comment (kazet) [inline app/src/worker.py:25]: if it is able to check email messages, it shouldn't be called like that -  why not just "scan_job" calling "scan_message_and_domain" or "scan_domain" ?

File: app/templates/check_running.html (hunk 3 above)
Comment (kazet) [reply app/templates/check_running.html:19]: it's still - need some help in ui design?

File: docker-compose.test.yml [hunk 7]
```diff
@@ -13,6 +13,18 @@ services:
...
       REDIS_URL: redis://redis-test:6379/0
     command: bash -c "/wait-for-it.sh db-test:5432 --  uvicorn src.app:app --host 0.0.0.0 --port 8000 --proxy-headers"
+  worker:
+    build:
+      context: .
+      dockerfile: app/docker/Dockerfile
+    environment:
+      APP_DOMAIN: "app"
+      DB_URL: postgresql+psycopg2://postgres:postgres@db-test:5432/mailgoose
+      FORWARDED_ALLOW_IPS: "*"
```
Comment (kazet) [inline docker-compose.test.yml:23]: is this one needed?

File: app/templates/check_running.html [hunk 8]
```diff
@@ -0,0 +1,20 @@
+{% extends "custom_layout.html" %}
+
+{% block header_additional %}
+    <meta http-equiv="refresh" content="5">
+{% endblock %}
+
+{% block body %}
+    <div class="container">
+        <div class="p-4">
+            <h1>{% trans %}Configuration analysis is running{% endtrans %}</h1>
```
Comment (kazet) [inline app/templates/check_running.html:10]: looking a lot nicer than the original!

**Review by kazet — APPROVED** [review]

**Pull request merged**